Src/FluentAssertions/Specialized/FunctionAssertions.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Specialized
+{
+    /// <summary>
+    /// Contains a number of methods to assert that a synchronous function yields the expected result.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class FunctionAssertions<T>
+    {
+        private readonly IExtractExceptions extractor;
+
+        public FunctionAssertions(Func<T> subject, IExtractExceptions extractor)
+        {
+            this.extractor = extractor;
+            Subject = subject;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Func{T}"/> that is being asserted.
+        /// </summary>
+        public Func<T> Subject { get; }
```

Comment:
I did not even analyze it, because all the other classes including `ReferenceTypeAssertions` have public `Subject` property. I would keep it public even for the sake of keeping a standard approach.